### PR TITLE
SLING-8934 Create new metric for the number of subscribers of distrib…

### PR DESCRIPTION
**Pull request by akrainiouk**
…ution publisher

### CODE REVIEW

File: src/main/java/org/apache/sling/distribution/journal/impl/publisher/DistributionPublisher.java [hunk 1]
```diff
@@ -167,6 +173,13 @@ public void activate(PublisherConfiguration config, BundleContext context) {
         
         String msg = String.format("Started Publisher agent %s with packageBuilder %s, queuedTimeout %s",
                 pubAgentName, pkgType, queuedTimeout);
+        if (metricsService != null) {
```
Comment (cschneider) [inline src/main/java/org/apache/sling/distribution/journal/impl/publisher/DistributionPublisher.java]: You can not use the metricsService in the Constructor. Instead this must happen in activator. As the @Reference is mandatory you also do not need to check for null. Instead you could use requireNonNull(..) to make sure we do not forget to set it during tests.

**Comment by tmaret** [comment]
Approach  LGTM, +1 with @cschneider findings.

File: src/main/java/org/apache/sling/distribution/journal/impl/publisher/DistributionPublisher.java (hunk 1 above)
Comment (akrainiouk) [reply src/main/java/org/apache/sling/distribution/journal/impl/publisher/DistributionPublisher.java]: This **is** and activator code, not sure were this concern about constructor is coming from.
metricsService field is indeed properly injected in real OSGI environment. The problem was in the unit test where it was remaining null. I think I found the root cause of it though, which is a name mismatch between the mock defined in the test and actual field name. I also missed the fact that there is already an existing reference to DistributionMetricsService available in DistributionPublisher under proper name. So I am committing the fix which will clean this duplication and at the same time make above null check unnecessary.
I wish I could also add unit test for the metric as well, but this requires much more extensive mocking so I am deferring it for later.

Comment (cschneider) [reply src/main/java/org/apache/sling/distribution/journal/impl/publisher/DistributionPublisher.java]: Alexei .. you are right.. it looked different in the diff view but when I look at the full file it is correct.

**Pull request merged**